File: Java/src/main/java/br/com/locaweb/locamail/api/utils/ListUtils.java
package br.com.locaweb.locamail.api.utils;

import br.com.locaweb.locamail.api.model.Convidado;
import br.com.locaweb.locamail.api.model.Email;
import br.com.locaweb.locamail.api.model.RespostaEmail;
import br.com.locaweb.locamail.api.repository.ConvidadoRepository;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class ListUtils {

    public static String listaParaString(List<String> lista) {
        return lista.stream()
                .collect(Collectors.joining(", "));
    }

    public static List<String> stringParaLista(String str) {
        if (str == null) {
            str = "";
        }
        return List.of(str.split(", "));
    }

    public static void atualizarTodosDestinatariosList(ArrayList<String> todosDestinatarios, Email email, List<RespostaEmail> respostaEmailList) {
        todosDestinatarios.addAll(stringParaLista(email.getDestinatario()));
        todosDestinatarios.addAll(stringParaLista(email.getCc()));
        todosDestinatarios.addAll(stringParaLista(email.getCco()));

        for (RespostaEmail resposta : respostaEmailList) {
            for (String destinatario : stringParaLista(resposta.getDestinatario())) {
                if (!todosDestinatarios.contains(destinatario)) {
                    todosDestinatarios.add(destinatario);
                }
            }

            for (String cc : stringParaLista(resposta.getCc())) {
                if (!todosDestinatarios.contains(cc)) {
                    todosDestinatarios.add(cc);
                }
            }

            for (String cco : stringParaLista(resposta.getCco())) {
                if (!todosDestinatarios.contains(cco)) {
                    todosDestinatarios.add(cco);
                }
            }
        }
    }

    public static List<Convidado> returnListConvidado(List<Long> listIdConvidado, ConvidadoRepository convidadoRepository) {
        List<Convidado> list = new ArrayList<>();
        for (Long id : listIdConvidado) {
            list.add(convidadoRepository.listarConvidadoPorId(id));
        }
        return list;
    }
}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/utils/ListUtils.kt
package br.com.fiap.locawebmailapp.utils

import androidx.compose.runtime.MutableState
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.snapshots.SnapshotStateList
import br.com.fiap.locawebmailapp.database.repository.ConvidadoRepository
import br.com.fiap.locawebmailapp.model.Convidado
import br.com.fiap.locawebmailapp.model.EmailComAlteracao
import br.com.fiap.locawebmailapp.model.IdConvidado
import br.com.fiap.locawebmailapp.model.RespostaEmail
import br.com.fiap.locawebmailapp.utils.api.callLocaMailApiListarConvidadoPorId

fun listaParaString(lista: List<String>): String {
    return lista.withIndex().joinToString(", ") { it.value }
}

fun stringParaLista(str: String): List<String> {
    return str.split(", ")
}

fun atualizarTodosDestinatariosList(todosDestinatarios: ArrayList<String>, email: EmailComAlteracao, respostaEmailList: List<RespostaEmail>){

    todosDestinatarios.addAll(stringParaLista(email.destinatario))
    todosDestinatarios.addAll(stringParaLista(email.cc))
    todosDestinatarios.addAll(stringParaLista(email.cco))

    for (resposta in respostaEmailList) {
        for (destinatario in stringParaLista(resposta.destinatario)) {
            if (!todosDestinatarios.contains(destinatario)) {
                todosDestinatarios.add(destinatario)
            }
        }

        for (cc in stringParaLista(resposta.cc)) {
            if (!todosDestinatarios.contains(cc)) {
                todosDestinatarios.add(cc)
            }
        }

        for (cco in stringParaLista(resposta.cco)) {
            if (!todosDestinatarios.contains(cco)) {
                todosDestinatarios.add(cco)
            }
        }
    }
}

fun returnListConvidado(
    listIdConvidado: List<Long>,
    isLoading: MutableState<Boolean>,
    isError: MutableState<Boolean>,
    listConvidado: SnapshotStateList<Convidado>
) {

    for (id in listIdConvidado) {

        callLocaMailApiListarConvidadoPorId(
            id,
            onSuccess = {
                convidadoRetornado ->
                if (convidadoRetornado != null) {
                    listConvidado.add(convidadoRetornado)
                }
            },
            onError = {
                isError.value = true
                isLoading.value = false
            }
        )
    }
}